(DOCSP-35339): Add docs release notes for w/e January 12 (#3153)
mongodb/docs-realm: docs-release-notes.md

# January 12, 2024

## C++ SDK

Prepare docs for C++ GA release, including:
- Remove "Preview" from naming.
- Move Bluehawked, tested code examples out of the experimental namespace, and generate updated examples.
- Remove tabs for "Current" and "Deprecated" (Beta and Alpha) syntaxes.
- Remove `.. versionadded::` and `.. versionchanged::` directives to reset for GA release.
- Update API reference links.
- Update the Google authentication examples for updated naming in v0.6.0-preview.
- Add new documentation and Bluehawked code examples for new functionality and API changes in GA:
  - Quick Start
    - New section about closing a realm.
    - Remove sync session initialization and method call from the "Open a Synced Realm" example.
  - Realm Files
    - Configure & Open a Realm: New section about closing a realm.
    - Reduce Realm File Size: New page about compacting a realm.
  - Application Services/Call a Function: Update the example to remove BSON and use the new string-based API.
  - Manage Users

>     - Authenticate Users: Update the Custom Function User example to remove BSON and use the new string-based API.

Reviewer: ```suggestion
    - Authenticate Users: Update the Custom Function User example to remove BSON with the new string-based API.
```

>     - Custom User Data: Update examples that show calling a Function to create, update, and delete user data to remove BSON and use the new string-based API.

Reviewer: ```suggestion
    - Custom User Data: Update examples that call a Function to create, update, and delete user data to remove BSON and use the new string-based API.
```
